Reneiski_Homework1/ReneiskiSelenium11/Program.cs

```diff
@@ -0,0 +1,246 @@
+            firstCheckBox.Click();
+
+            var checkboxIsChecked = _driver.FindElement(By.XPath("//*[@class='rct-icon rct-icon-check']"));
+            string attributeClass = checkboxIsChecked.GetAttribute("class");
+            Assert.IsTrue(attributeClass.Contains("rct-icon rct-icon-check"));
+
+            var currentResult = _driver.FindElement(By.Id("result"));
+            var currentResultText = currentResult.Text;
+
+            Assert.IsTrue(currentResultText.Contains("You have selected :"));
+            Assert.IsTrue(currentResultText.Contains("home"));
+            Assert.IsTrue(currentResultText.Contains("desktop"));
+            Assert.IsTrue(currentResultText.Contains("notes"));
+            Assert.IsTrue(currentResultText.Contains("commands"));
+            Assert.IsTrue(currentResultText.Contains("documents"));
+            Assert.IsTrue(currentResultText.Contains("workspace"));
+            Assert.IsTrue(currentResultText.Contains("react"));
+            Assert.IsTrue(currentResultText.Contains("angular"));
+            Assert.IsTrue(currentResultText.Contains("veu"));
+            Assert.IsTrue(currentResultText.Contains("office"));
+            Assert.IsTrue(currentResultText.Contains("public"));
+            Assert.IsTrue(currentResultText.Contains("private"));
+            Assert.IsTrue(currentResultText.Contains("classified"));
+            Assert.IsTrue(currentResultText.Contains("general"));
+            Assert.IsTrue(currentResultText.Contains("downloads"));
+            Assert.IsTrue(currentResultText.Contains("wordFile"));
+            Assert.IsTrue(currentResultText.Contains("excelFile"));
+        }
+
+        [Test]
+        public void SecondRadioButtonTest()
+        {
+            _driver.Navigate().GoToUrl("https://demoqa.com/radio-button");
+
+            var radioButtonYes = _driver.FindElement(By.XPath("//*[@for='yesRadio']"));
+            radioButtonYes.Click();
+
+            var currentResult = _driver.FindElement(By.XPath("//*[@class='mt-3']"));
+            var currentResultText = currentResult.Text;
+
+            Assert.IsTrue(currentResultText.Contains("You have selected "));
+            Assert.IsTrue(currentResultText.Contains("Yes"));
+
+            var radioButtonImpressive = _driver.FindElement(By.XPath("//*[@for='impressiveRadio']"));
+            radioButtonImpressive.Click();
+
+            var currentResult1 = _driver.FindElement(By.XPath("//*[@class='mt-3']"));
+            var currentResultText1 = currentResult1.Text;
+
+            Assert.IsTrue(currentResultText1.Contains("You have selected "));
+            Assert.IsTrue(currentResultText1.Contains("Impressive"));
+        }
+
+        [Test]
+        public void ThirdWebTablesTest()
+        {
+            _driver.Navigate().GoToUrl("https://demoqa.com/webtables");
+
+            var currentResult = _driver.FindElement(By.XPath("//*[@class='rt-table']"));
```

Comment:
You checked whether data exists in the table, but it's better to make this check deeper: you have Cierra record hardcoded in your site, it's always the first row. So, it's better to make this check like check whether the first row contains expected data  + check it by column - find the first row and cell that corresponds to the First Name column, get text, compare text then find a cell that corresponds to first row (as Cierra record is hardcoded in the first row), find cell that corresponds to Last Name columnd, get text, compare text with expected etc

Reply:
Fixed